### PR TITLE
Fix memory initialization errors

### PR DESCRIPTION
**Pull request by TooAngel**

### CODE REVIEW

**Comment by TooAngel** [comment]
DCBOT: 
This repository is under [democratic collaboration](https://github.com/TooAngel/democratic-collaboration) and will be merged automatically.

The merge decision is based on the outcome of the reviews:
 - `Approve` add the reviewer value (number of commits) to the `metric`
 - `Request changes` substract the reviewer value from the `metric`

 - `metric > 99%` merge now
 - `metric > 75%` merge in 1 day
 - `metric > 50%` merge in 3 days
 - `metric >= 0%` merge in 7 days

Please review the PR to make a good democratic decision.

This is the placeholder for reviewer summoning :-) (top two + random)

File: src/prototype_room_basebuilder.js [hunk 1]
```diff
@@ -259,7 +259,7 @@ Room.prototype.setupStructure = function(structure) {
   }
 
   var max = CONTROLLER_STRUCTURES[structure][this.controller.level];
-  for (let pos of (this.memory.position.structure[structure] || [])) {
+  for (let pos of(this.memory.position.structure[structure] || [])) {
```
Comment (Somotaw) [inline src/prototype_room_basebuilder.js]: add whitespace back? :)

**Comment by TooAngel** [comment]
DCBOT: Current status coefficient: 0.519434628975 votes: 147 total: 283 age: 0

**Comment by TooAngel** [comment]
DCBOT: Code update recognized, countdown starts fresh.

File: src/visualizer.js [hunk 2]
```diff
@@ -64,7 +64,7 @@ if (config.visualizer.enabled) {
           });
         }
         // draw creep positions
-        if (config.visualizer.showCreeps) {
+        if (config.visualizer.showCreeps && room.memory.position) {
```
Comment (Somotaw) [inline src/visualizer.js]: sure we need these? if vusual runs on a room the positions whould already be in place?

**Comment by TooAngel** [comment]
DCBOT: Current status coefficient: 0.519434628975 votes: 147 total: 283 age: 0

File: src/prototype_room_basebuilder.js (hunk 1 above)
Comment (TooAngel) [inline src/prototype_room_basebuilder.js]: hm, one of my formatter does that. Will check

**Comment by TooAngel** [comment]
DCBOT: Current status coefficient: 0.519434628975 votes: 147 total: 283 age: 0

**Comment by TooAngel** [comment]
DCBOT: Current status coefficient: 0.597173144876 votes: 169 total: 283 age: 0

File: src/visualizer.js (hunk 2 above)
Comment (TooAngel) [inline src/visualizer.js]: I just did `delete Memory.rooms` which could be the reason. But we should handle these cases. In the past there were also Memory issues on the server and we want to be able to recover.

**Comment by TooAngel** [comment]
DCBOT: Current status coefficient: 0.597173144876 votes: 169 total: 283 age: 0

File: src/prototype_room_basebuilder.js (hunk 1 above)
Comment (TooAngel) [inline src/prototype_room_basebuilder.js]: `jsbeautifier` makes this changed based on the settings in the repository. So in the end it is 'fine' (also couldn't find a quick way to change the configuration according :-))

File: src/visualizer.js (hunk 2 above)
Comment (Somotaw) [inline src/visualizer.js]: Aight :)

**Comment by TooAngel** [comment]
DCBOT: Current status coefficient: 0.597173144876 votes: 169 total: 283 age: 0

**Comment by TooAngel** [comment]
DCBOT: Current status coefficient: 0.597173144876 votes: 169 total: 283 age: 0

File: src/prototype_room_basebuilder.js (hunk 1 above)
Comment (Somotaw) [inline src/prototype_room_basebuilder.js]: Okay :D

**Comment by TooAngel** [comment]
DCBOT: Current status coefficient: 0.597173144876 votes: 169 total: 283 age: 0

**Comment by TooAngel** [comment]
DCBOT: Current status coefficient: 0.766784452297 votes: 217 total: 283 age: 0

**Comment by TooAngel** [comment]
DCBOT: Current status coefficient: 0.766784452297 votes: 217 total: 283 age: 0

**Comment by TooAngel** [comment]
DCBOT: Current status coefficient: 0.766784452297 votes: 217 total: 283 age: 0

File: src/config_brain_memory.js [hunk 3]
```diff
@@ -47,7 +47,10 @@ brain.prepareMemory = function() {
   for (let name in Memory.creeps) {
     if (!Game.creeps[name]) {
       let role = Memory.creeps[name].role;
-      let roleStat = Memory.stats[userName].roles[role];
+      let roleStat;
+      if (!Memory.stats[userName]) {
```
Comment (WhiteHalmos) [inline src/config_brain_memory.js]: Remove !

Comment (Somotaw) [inline src/config_brain_memory.js]: What is the issue here WhiteHalmos?

**Comment by TooAngel** [comment]
@Nevrdid are these errors already addressed in your PR https://github.com/TooAngel/screeps/pull/145 and better?

File: src/config_brain_memory.js (hunk 3 above)
Comment (Nevrdid) [inline src/config_brain_memory.js]: if Memory.stats[username] is undefined then roleStat = Memory.stats[userName].roles[role]; is undefined too

**Comment by TooAngel** [comment]
DCBOT: A new review, yeah.

    Votes: 41/285
    Coefficient: 0.143859649123
    Days to merge: 8.56140350877
    Age: 1

**Comment by Nevrdid** [comment]
yeah, it should be fixed as i declare username from spawn if memory.username not declared.
about stats.roles it is created when stas is enabled.
Somotaw have issue with stats.roles not defined but it shouldn't happen as i tryed to launch it from clean memory and all was corectly setup without error. so don't understand. and i don't wanna add check for things which is defined for sure.

**Pull request closed without merging**